src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,142 @@
+package picard.sam;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordIterator;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.OverlapDetector;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class AddOATagTest {
+    @DataProvider(name="testOAData")
```

Comment:
Maybe instead of the three separated test cases, a single one containing all of them is enought unless there is a specific reason for separate interval/overwrite tests. In that case, a single test method will has an input file, overwrite boolean, interval list (or null if none) and expected output. This will allow to share expected files and input files (e.g., the `aligned_withoutOA.sam` and `aligned_withOtherOA.sam` as inputs should produce both the same `expected_overwriteOA.sam` if they contain the same records), and be a good integration test to check if they are really the same.

Reply:
I left these as separate tests, but now they are all file based integration tests. Each case has an expected sam file.